framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -230,10 +230,14 @@
    * command line.
    */
   private final boolean checkPurity;
+  // TODO: Given that setting this to false is problematic, should it be removed entirely?
```

Comment:
@smillst I am inclined to remove the variable.  What do you think?

Reply:
I removed it.  I think the Lock Checker always set it.

Why was `getCheckerNames()` nondeterministic?  Was it because WPI was creating file `META-INF/services/javax.annotation.processing.Processor` nondeterministically?